IQSS/dataverse
# #9074 - Added support for building sphinx docs using python 3.10+

**What this PR does / why we need it**:

Add support for building Sphynx guides using Python 3.10+

**Which issue(s) this PR closes**:

- Closes #9074

**Special notes for your reviewer**:

Tested Sphynx version 5.3.0 with Python 3.9.7 and the docs were built successfully. However, chose to add rules for Python >= 3.10 and < 3.10 to ensure backward compatibility

**Suggestions on how to test this**:

- Install Sphynx requirements while using Python 3.9.x and build guides: Sphynx version 3.5.4 should be installed and build should be successful
- Install Sphynx requirements while using Python 3.10.x and build guides: Sphynx version 5.3.0 should be installed and build should be successful
